.github/workflows/test.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  unit_tests:
+    name: Unit Tests
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version: ['3.8', '3.9']
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Setup python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: x64
+
+    - name: Install pip dependencies
+      run: |
+       python -m pip install flake8 flake8-blind-except flake8-builtins flake8-class-newline flake8-comprehensions flake8-deprecated flake8-docstrings flake8-import-order flake8-quotes pytest
+       python -m pip freeze
```

Comment:
Just curious, what does this achieve?

Also, I know this is typically handled with `rosdep` but for this workflow's benefit, would a `requirements.txt` file containing the `pip freeze` for this project be a good idea?

Reply:
The idea with the `pip freeze` is really just to print out the versions of all of the pip packages are being installed.  It's useful if an error occurs and you are trying to recreate the environment locally.

`pip list` would do as well, but we tend to prefer `pip freeze` in our infrastructure for reasons I don't fully understand.

Reply:
Switching to requirements.txt is a good idea.  I've done this in 7f8d743.